src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/begrunnelser/BegrunnelserForPeriodeContext.kt

```diff
@@ -71,9 +75,21 @@ class BegrunnelserForPeriodeContext(
     private fun Begrunnelse.triggesForVedtaksperiode(): Boolean {
         val sanityBegrunnelse = this.tilSanityBegrunnelse(sanityBegrunnelser) ?: return false
 
+        if (sanityBegrunnelse.endretUtbetalingsperiode.isNotEmpty()) return erEtterEndretPeriodeAvSammeÅrsak(
+            sanityBegrunnelse
+        )
+
         return hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(this, sanityBegrunnelse).isNotEmpty()
     }
 
+    private fun erEtterEndretPeriodeAvSammeÅrsak(begrunnelse: SanityBegrunnelse) =
+        endretUtbetalingsandeler.any { endretUtbetalingAndel ->
+            endretUtbetalingAndel.tom?.sisteDagIInneværendeMåned()
+                ?.erDagenFør(utvidetVedtaksperiodeMedBegrunnelser.fom) ?: false &&
+                personResultater.any { person -> person.aktør.aktørId == endretUtbetalingAndel.person?.aktør?.aktørId } &&
+                begrunnelse.endringsårsaker.contains(endretUtbetalingAndel.årsak)
```

Comment:
Litt mye på en gang her. Tror det hadde vært digg med noen oppklarende variabler:

```suggestion
            val endringsperiodeErDagenEtterVedtaksperiode = 
                endretUtbetalingAndel.tom?.sisteDagIInneværendeMåned()
                ?.erDagenFør(utvidetVedtaksperiodeMedBegrunnelser.fom) ?: false
            
            val endringsperiodeGjelderSammePersonSomVedtaksperiode = 
                personResultater.any { person -> person.aktør.aktørId == endretUtbetalingAndel.person?.aktør?.aktørId } 
            
            val begrunnelseHarSammeÅrsakSomEndringsperiode = 
                personResultater.any { person -> person.aktør.aktørId == endretUtbetalingAndel.person?.aktør?.aktørId } 

            endringsperiodeErDagenEtterVedtaksperiode &&
                endringsperiodeGjelderSammePersonSomVedtaksperiode &&
                begrunnelseHarSammeÅrsakSomEndringsperiode
```

Reply:
Enig! Kjører på med dette, endrer dog på `begrunnelseHarSammeÅrsakSomEndringsperiode`.